### PR TITLE
Switching to Webpack

### PR DESCRIPTION
**Pull request by tgilan**
## Switching Data Lasso build process to webpack
There are multiple reasons for this:
- Stylesheet build task watches were broken
- Shaders required a workaround, where they were inline scripts rendered on the page with `type="x-shader/x-vertex` and then picked up; Now shaders are their own `.glsl` files that are consumed by `shader-loader`

### What's inside
- Webpack core library, loaders for SASS, shaders, ES6 and jsx
- There are now only two tasks: `npm start` and `npm run build`:
  - `npm start` starts webpack dev server with hotloading enabled on same port as before
  - `npm run build` builds minified script for distribution. It also sets `NODE_ENV` to `production` so that React switches from development mode
- Removed anything that related to gulp - gulp tasks, utils and transforms
- Removed shaders workaround; shaders are now in `.glsl`
- Removed usage jQuery for DOM manipulation, as it gave problems with integration
- Made the link to test dataset protocol-agnostic

That should be the final change before Data Lasso will be bumped to **2.0.0**.

@heylookltsme @kmck

### CODE REVIEW

File: src/components/AxisControls/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 
+var $ = require('jquery');
```
Comment (tgilan) [inline src/components/AxisControls/index.js]: I should probably unify usages of var vs. const across the app - will add it to todo list.

File: src/graph/index.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ var SelectionHelper = require('./helpers/selection');
 var axisGeometry = require('./geometry/axis');
 var Mouse = require('./helpers/mouse');
 var Keyboard = require('./helpers/keyboard');
-var shaders = require('../templates/shaders.tpl');
+var vertexShader = require('./shaders/vertex.glsl');
+var fragmentShader = require('./shaders/fragment.glsl');
```
Comment (tgilan) [inline src/graph/index.js]: ❤️

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                },
+            },
+        ],
+    },
+    devtool: production ? '' : 'inline-source-map',
+    devServer: {
+        contentBase: path.join(__dirname, 'public'),
+    },
+    sassLoader: {
+        sourcemap: !production,
```
Comment (kmck) [inline webpack.config.js]: i think this is supposed to be `sourceMap`

**Comment by heylookltsme** [comment]
Are these cool yellow warnings expected - 

2__bash

**Comment by heylookltsme** [comment]
omg hot loading is _sick_. we need that on our roadmap.

**Comment by tgilan** [comment]
Yes, those are expected - Uglify removes dead code (!) and displays a warning about it. Those can be hidden with `compress: {warnings: false}` but I like to see what was removed, especially since that will only be on production builds, which should happen only with version bumps.

**Comment by heylookltsme** [comment]
Tested; works great 👍

**Comment by kmck** [comment]
👍

**Pull request merged**